File: Shop/src/jsp/Validate.java
package jsp;

import java.text.SimpleDateFormat;

public class Validate {
	// Functions to validate the values by passing it to validate functions
	public static String validateEmail(String email) {
		String error = "";
		if (email.isEmpty()) {
			error = "This field is required.";
		} else if (email.length() > 32) {
			error = "This field cannot exceed the length of 32.";
		}
		// w=word, + = more then 1, * = optional {2,4} = 2-4 words
		else if (!email.matches("^(\\w+(\\.\\w+)*@\\w+\\.\\w{2,4})$")) {
			error = "Invalid email.";
		}
		return error;
	}

	public static String validatePassword(String password) {
		String error = "";
		if (password.isEmpty()) {
			error = "This field is required.";
		} else if (password.length() > 24) {
			error = "This field cannot exceed the length of 24.";
		}
		return error;
	}

	public static String validateBirthday(String birthday) {
		String error = "";
		if (birthday.isEmpty()) {
			error = "This field is required.";
		} else if (!birthday.matches("^(\\d{2}/\\d{2}/\\d{4})$")) {
			error = "Invalid date format, Expected MM/dd/yyyy.";
		} else {
			try {
				java.util.Date d = new SimpleDateFormat("MM/dd/yyyy").parse(birthday);
				if (d.after(new java.util.Date())) {
					error = "Invalid Date.";
				}
			} catch (Exception e) {
				error = "Invalid Date.";
			}

		}
		return error;
	}

}

File: Shop/WebContent/js/validate.js
var email_regex = /^(\w+(\.\w+)*@\w+\.\w{2,4})$/, birthday_regex = /^(\d{2}\/\d{2}\/\d{4})$/;
function validateEmail(e) {
	e = e.target.value;
	var error = "";
	if (e.length == 0) {
		error = "This field is required.";
	} else if (e.length > 32) {
		error = "This field cannot exceed the length of 32.";
	}
	else if (!email_regex.test(e)) {
		error = "Invalid email.";
	}
	$("#email_error").html(error);
	//isAllValid();
}

function validatePassword(e) {
	e = e.target.value;
	var error = "";
	if (e.length == 0) {
		error = "This field is required.";
	} else if (e.length > 24) {
		error = "This field cannot exceed the length of 24.";
	}
	$("#password_error").html(error);
	//isAllValid();
}
function validateBirthday(e) {
	e = e.target.value;
	var error = "";
	if (e.length == 0) {
		error = "This field is required.";
	} else if (!birthday_regex.test(e)) {
		error = "Invalid date format, Expected MM/dd/yyyy.";
	} else {
		var date = new Date(e), d = date.getDate(), m = date.getMonth() + 1, y = date
				.getFullYear();
		if (d < 10) {
			d = '0' + d;
		}
		if (m < 10) {
			m = '0' + m;
		}
		d = m + "/" + d + "/" + y;
		if (d != e || (date > new Date())) {
			error = "Invalid date.";
		}
	}
	$("#birthday_error").html(error);
	//isAllValid();
}